control-spec.txt

```diff
@@ -1537,14 +1537,20 @@
     "HSFETCH" SP (HSAddress / "v" Version "-" DescId)
               *[SP "SERVER=" Server] CRLF
 
-    HSAddress = 16*Base32Character
-    Version = "2"
+    HSAddress = 16*Base32Character / 56*Base32Character
+    Version = "2" / "3"
     DescId = 32*Base32Character
     Server = LongName
 
   This command launches hidden service descriptor fetch(es) for the given
   HSAddress or DescId.
 
+  HSAddress can be version 2 or version 3 addresses. DescIDs can only be
+  version 2 IDs. Version 2 addresses consist of 16*Base32Character and
+  version 3 addresses consist of 56*Base32Character. The controller MUST
+  assert that the HSAddress is 16 characters long if something prefixed
+  with "v2-" is specified.
```

Comment:
This is wrong actually. It was my mistake... We need to remove that last sentence. As you said, anything "vX-" is only for the descriptor ID.